src/de.rs

```diff
@@ -283,7 +283,11 @@ impl FromStr for RawHrp {
 		let si_prefix: Option<SiPrefix> = if parts.2.is_empty() {
 			None
 		} else {
-			Some(parts.2.parse()?)
+			let si = parts.2.parse()?;
+			if amount.checked_mul(si.multiplier()).is_non() {
```

Comment:
Nice catch, I totally oversaw this possible overflow.

Reply:
Lol, I guess it'd be a nicer catch if I could spell is_none.